packages/zwave-js/src/lib/commandclass/EntryControlCC.ts

```diff
@@ -0,0 +1,683 @@
+@expectedCCResponse(EntryControlCCKeySupportedReport)
+export class EntryControlCCKeySupportedGet extends EntryControlCC {}
+
+@CCCommand(EntryControlCommand.EventSupportedReport)
+export class EntryControlCCEventSupportedReport extends EntryControlCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		validatePayload(this.payload.length >= 6);
+
+		const dataTypeLength = this.payload[0] & 0b0000_0011;
+		validatePayload(this.payload.length >= 6 + dataTypeLength);
+		this._dataTypeSupported = parseBitMask(
+			this.payload.slice(1, 1 + dataTypeLength),
+			0,
+		);
+
+		const eventTypeLength = this.payload[1 + dataTypeLength] & 0b0001_1111;
+		validatePayload(
+			this.payload.length >= 6 + dataTypeLength + eventTypeLength,
+		);
+		this._eventTypeSupported = parseBitMask(
+			this.payload.slice(
+				2 + dataTypeLength,
+				2 + dataTypeLength + eventTypeLength,
+			),
+			0,
+		);
+
+		const minMaxStart = 2 + dataTypeLength + eventTypeLength;
+		this._keyCachedSizeMin = this.payload[minMaxStart];
+		this._keyCachedSizeMax = this.payload[minMaxStart + 1];
+		this._keyCachedTimeoutMin = this.payload[minMaxStart + 2];
+		this._keyCachedTimeoutMax = this.payload[minMaxStart + 3];
+		this.persistValues();
+	}
+
+	private _dataTypeSupported: EntryControlDataTypes[];
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnly,
+		type: "number[]",
+		label: "Supported data types",
+		description: "A list of supported data types",
+	})
+	public get dataTypeSupported(): EntryControlDataTypes[] {
+		return this._dataTypeSupported;
+	}
+
+	private _eventTypeSupported: EntryControlEventTypes[];
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnly,
+		type: "number[]",
+		label: "Supported event types",
+		description: "A list of supported event types",
+	})
+	public get eventTypeSupported(): EntryControlEventTypes[] {
+		return this._eventTypeSupported;
+	}
+
+	private _keyCachedSizeMin: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Minimum supported value of key cache size",
+	})
+	public get keyCachedSizeMin(): number {
+		return this._keyCachedSizeMin;
+	}
+
+	private _keyCachedSizeMax: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Maximum supported value of key cache size",
+	})
+	public get keyCachedSizeMax(): number {
+		return this._keyCachedSizeMax;
+	}
+
+	private _keyCachedTimeoutMin: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Minimum supported value of key cache timeout",
+	})
+	public get keyCachedTimeoutMin(): number {
+		return this._keyCachedTimeoutMin;
+	}
+
+	private _keyCachedTimeoutMax: number;
+
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyUInt8,
+		label: "Maximum supported value of key cache timeout",
+	})
+	public get keyCachedTimeoutMax(): number {
+		return this._keyCachedTimeoutMax;
+	}
```

Comment:
These shouldn't be values, but rather change the min/max of the key cache size/timeout value metadata.

Reply:
Also regarding the naming:
`minKeyCacheSize` etc sounds more natural.

Reply:
How do I make them value metadata instead of value?

Reply:
How do I make them value metadata instead of value?

Reply:
An example can be found in `ThermostatFanModeCCSupportedReport` where the states are changed to match what the device reports.
You can use the same approach to change min/max of the value's corresponding metadata.